### ISSUE TITLE
Uncaught TypeError: Cannot read property 'global' of undefined | Contact Form Bug

### ISSUE THREAD

**Issue by mcrowell** [elementor/elementor#13809]
## Prerequisites

- [x] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [x] The issue still exists against the latest stable version of Elementor.

## Description

There's a styling bug on the backend when using the contact form with MailerLite.

## Steps to reproduce

Add a contact form widget. Activate a MailerLite API key. Try to adjust any style element and you will see errors in the browser Console. 

Attached is a screenshot of the Console errors.
![Mailerlite-Form-Bug](https://user-images.githubusercontent.com/7768009/106934328-fcd18480-66e7-11eb-9dcd-31c8aeacba56.jpg)

Uncaught TypeError: Cannot read property 'global' of undefined
TypeError: Cannot read property 'global' of undefined

## Isolating the problem
- [x] This bug happens with only Elementor plugin active (and Elementor Pro).
- [x] This bug happens with a default WordPress theme active.
- [x] I can reproduce this bug consistently using the steps above.

## Environment

[system-info-wickydemo.com-04-02-2021.txt](https://github.com/elementor/elementor/files/5927574/system-info-wickydemo.com-04-02-2021.txt)

**Comment by rafiul** [comment]
I am facing the same problem after update to 3.1.1 console error on screenshot given below. 

![error](https://user-images.githubusercontent.com/6370697/106995889-13d8ab00-67aa-11eb-874b-676440c819e9.jpg)

**Comment by ppulpipatnan** [comment]
I am having the same issue on Elementor 3.1.1

**Comment by kevtanner** [comment]
Same issue

**Comment by Ninetheme** [comment]
Same issue with [Awam](https://themeforest.net/item/awam-portfolio-wordpress-theme/28176962) Theme

**Comment by AleksBaikalski** [comment]
+1

**Comment by akashxyz** [comment]
happening with repeater field too.

**Comment by FrancescoMasaia** [comment]
+1

**Comment by francescodicandia** [comment]
+1

**Comment by elMui** [comment]
+1

**Comment by FrancescoMasaia** [comment]
solved by downgrading to 2.9.9 version

**Comment by markbosky** [comment]
+1

**Comment by mcrowell** [comment]
@shilo-ey is this bug going to be fixed in an upcoming release? Or should we roll back until future notice?

**Comment by ZsofiaMajoros** [comment]
+1

**Comment by zovdienko** [comment]
Have same problems only with widgets with Repeater control type. 

            [
                'type' => Controls_Manager::REPEATER,
                'show_label' => true,
                'fields' => array_values( $slide_repeater->get_controls() ),
                'title_field' => '{{{ slide_title }}}',
            ]


Remove **_array_values()_** in **_'fields'_** fix my problem.

`'fields' => $slide_repeater->get_controls(),`

**Comment by rafiul** [comment]
@zovdienko  Thank you for finding out the exact problem. Now working perfectly.

**Comment by themerevealed** [comment]
@zovdienko Thanks a lot. Now everything thing is working properly.

**Comment by Canos** [comment]
Downgrading to 2.9.14 elementor plugin fixes the issue. Annoying but works.

**Comment by Mrityunjoy14** [comment]
I am facing the same problem.. I'm using elementor pro as well with jet element. I tested it with 2.9.14 version and its load fine but its unable to load jet elements which are very important for my site. :( please help

**Comment by mircobabini** [comment]
Same here.

**Comment by blocksera** [comment]
We have checked `elementor\includes\widgets\accordion.php` and replicated the repeater control code style. The issue is then fixed.

Original:
```php
$this->add_control(
	'buttons',
	[
		'label' => __('Buttons', 'emage-hover-effects'),
		'type' => Controls_Manager::REPEATER,
		'fields' => [
			[
				'name' => 'text',
				'label' => __( 'Text', 'emage-hover-effects'),
				'type' => Controls_Manager::TEXT
			]
		],
		'prevent_empty' => false,
		'title_field' => '{{{ text }}}',
	]
);
```

Fixed:
```php
$repeater = new Repeater();

$repeater->add_control(
	'text',
	[
		'label' => __( 'Text', 'emage-hover-effects'),
		'type' => Controls_Manager::TEXT,
		'label_block' => true,
		'default' => 'Button',
		'dynamic' => [
			'active' => true,
			'categories' => [TagsModule::TEXT_CATEGORY],
		]
	]
);

$this->add_control(
	'buttons',
	[
		'label' => __('Buttons', 'emage-hover-effects'),
		'type' => Controls_Manager::REPEATER,
		'fields' => $repeater->get_controls(),
		'prevent_empty' => false,
		'title_field' => '{{{ text }}}',
	]
);
```

**Comment by artifex-media** [comment]
Same here. Still present.

**Comment by mckenzieja** [comment]
This issue is still present when using forms.

**Comment by Zexler** [comment]
+1
[system-info-megalan.net.ua-20-04-2021.txt](https://github.com/elementor/elementor/files/6341673/system-info-megalan.net.ua-20-04-2021.txt)

**Comment by razon-xs** [comment]
Most of the time I have faced this issue for Repeater Controls. 
The correct way to call a Repeater Control:

```
$repeater = new \Elementor\Repeater();

$repeater->add_control(
	'list_title', [
		'label' => __( 'Title', 'plugin-domain' ),
		'type' => \Elementor\Controls_Manager::TEXT,
		'default' => __( 'List Title' , 'plugin-domain' ),
		'label_block' => true,
	]
);

$repeater->add_control(
	'list_content', [
		'label' => __( 'Content', 'plugin-domain' ),
		'type' => \Elementor\Controls_Manager::WYSIWYG,
		'default' => __( 'List Content' , 'plugin-domain' ),
		'show_label' => false,
	]
);

$repeater->add_control(
	'list_color',
	[
		'label' => __( 'Color', 'plugin-domain' ),
		'type' => \Elementor\Controls_Manager::COLOR,
		'selectors' => [
			'{{WRAPPER}} {{CURRENT_ITEM}}' => 'color: {{VALUE}}'
		],
	]
);

$this->add_control(
	'list',
	[
		'label' => __( 'Repeater List', 'plugin-domain' ),
		'type' => \Elementor\Controls_Manager::REPEATER,
		'fields' => $repeater->get_controls(),
		'default' => [
			[
				'list_title' => __( 'Title #1', 'plugin-domain' ),
				'list_content' => __( 'Item content. Click the edit button to change this text.', 'plugin-domain' ),
			],
			[
				'list_title' => __( 'Title #2', 'plugin-domain' ),
				'list_content' => __( 'Item content. Click the edit button to change this text.', 'plugin-domain' ),
			],
		],
		'title_field' => '{{{ list_title }}}',
	]
);
```

Official Documentation:  https://developers.elementor.com/elementor-controls/repeater-control/

**Comment by owaisahmed5300** [comment]
@zovdienko  Solution works. I had an old snippet of Elementor code that causing this. Thanks

**Comment by HeyMehedi** [comment]
This issue is still present.

**Comment by Squeezemind** [comment]
The issue is still present. Downgrade to 3.0.16 solves the problem. Please fix it

**Comment by cjchamado** [comment]
I tried in every way to change the element's version settings to resolve the error, but without success.

The solution in my case was to search the code for the problem and implement a fix in the script:

**wp-content/plugins/elementor/assets/js/editor-modules.js L:1990-1994**

```js
var globalControl = controls[controlGlobalKey];

if (!((_globalControl$global = globalControl.global) !== null && _globalControl$global !== void 0 && _globalControl$global.active)) {
      return false;
}
```

Checking in the logical check, if the variable "globalControl" really received a value, and only later access its "global" property.

```js
var globalControl = controls[controlGlobalKey];

if (!((_globalControl$global = typeof globalControl === 'object' ? globalControl.global : null) !== null && _globalControl$global !== void 0 && _globalControl$global.active)) {
      return false;
}
```

**If the change occurs in the .min file (editor-modules.min.js)**

Search for the snippet:

```js
null===(o=r[a].global

```

And replace with:

```js
null===(o=r[a]?r[a].global:null)
````

In short, a check to see if "globalControl" is an object and if it has the property "global" solves the problem in the script.

I reiterate that this was the palliative solution I found for my problem, which may not be the correct one or entirely in accordance with the intention of the check, I am not a deep connoisseur of WP or Elementor, but I solved my problem with this as a last approach .

Thanks!

**Comment by Sect0R** [comment]
@cjchamado , you lost the round bracket. Btw - Thanks!

> Search for the snippet:
> 
> null===(o=r[a].global
> And replace with:
> 
> null===(o=r[a]?r[a].global:null)

**Correct:**

Search for the snippet:
```js
null===(o=r[a].global)
```

**Comment by meksone** [comment]
Still open on version 3.8.1:
the fix suggested by @cjchamado works like a charm with an issue with **Premium Addons Pro** plugin.

**Comment by AbanobAkram** [comment]
If you are having this issue with **Premium Addons** plugin, you can add the code below in your theme functions.php file to fix it.

`add_filter( 'pa_display_conditions_values', function( $apply ) {
    return false;
});`

**Comment by mapsmarketing** [comment]
Can confirm this is an issue with Elementor itself even for v3.10.3 (Pro) and v3.10.2 (Free). The below solution by @razon-xs fixed this issue. The bug even prevents both styles and handlers being applied to the widget.

> Most of the time I have faced this issue for Repeater Controls. The correct way to call a Repeater Control:
> 
> ```
> $repeater = new \Elementor\Repeater();
> 
> $repeater->add_control(
> 	'list_title', [
> 		'label' => __( 'Title', 'plugin-domain' ),
> 		'type' => \Elementor\Controls_Manager::TEXT,
> 		'default' => __( 'List Title' , 'plugin-domain' ),
> 		'label_block' => true,
> 	]
> );
> 
> $repeater->add_control(
> 	'list_content', [
> 		'label' => __( 'Content', 'plugin-domain' ),
> 		'type' => \Elementor\Controls_Manager::WYSIWYG,
> 		'default' => __( 'List Content' , 'plugin-domain' ),
> 		'show_label' => false,
> 	]
> );
> 
> $repeater->add_control(
> 	'list_color',
> 	[
> 		'label' => __( 'Color', 'plugin-domain' ),
> 		'type' => \Elementor\Controls_Manager::COLOR,
> 		'selectors' => [
> 			'{{WRAPPER}} {{CURRENT_ITEM}}' => 'color: {{VALUE}}'
> 		],
> 	]
> );
> 
> $this->add_control(
> 	'list',
> 	[
> 		'label' => __( 'Repeater List', 'plugin-domain' ),
> 		'type' => \Elementor\Controls_Manager::REPEATER,
> 		'fields' => $repeater->get_controls(),
> 		'default' => [
> 			[
> 				'list_title' => __( 'Title #1', 'plugin-domain' ),
> 				'list_content' => __( 'Item content. Click the edit button to change this text.', 'plugin-domain' ),
> 			],
> 			[
> 				'list_title' => __( 'Title #2', 'plugin-domain' ),
> 				'list_content' => __( 'Item content. Click the edit button to change this text.', 'plugin-domain' ),
> 			],
> 		],
> 		'title_field' => '{{{ list_title }}}',
> 	]
> );
> ```
> 
> Official Documentation: https://developers.elementor.com/elementor-controls/repeater-control/

**Comment by FauzanEdris** [comment]
issue is still there with fields array

**Comment by Danieljarto** [comment]
> Can confirm this is an issue with Elementor itself even for v3.10.3 (Pro) and v3.10.2 (Free). The below solution by @razon-xs fixed this issue. The bug even prevents both styles and handlers being applied to the widget.
> 
> I highly recommend putting in a note on the documentation to let folks know about this, since the bug had been reported 2 years ago: https://developers.elementor.com/docs/editor-controls/control-repeater/
> 
> > Most of the time I have faced this issue for Repeater Controls. The correct way to call a Repeater Control:
> > ```
> > $repeater = new \Elementor\Repeater();
> > 
> > $repeater->add_control(
> > 	'list_title', [
> > 		'label' => __( 'Title', 'plugin-domain' ),
> > 		'type' => \Elementor\Controls_Manager::TEXT,
> > 		'default' => __( 'List Title' , 'plugin-domain' ),
> > 		'label_block' => true,
> > 	]
> > );
> > 
> > $repeater->add_control(
> > 	'list_content', [
> > 		'label' => __( 'Content', 'plugin-domain' ),
> > 		'type' => \Elementor\Controls_Manager::WYSIWYG,
> > 		'default' => __( 'List Content' , 'plugin-domain' ),
> > 		'show_label' => false,
> > 	]
> > );
> > 
> > $repeater->add_control(
> > 	'list_color',
> > 	[
> > 		'label' => __( 'Color', 'plugin-domain' ),
> > 		'type' => \Elementor\Controls_Manager::COLOR,
> > 		'selectors' => [
> > 			'{{WRAPPER}} {{CURRENT_ITEM}}' => 'color: {{VALUE}}'
> > 		],
> > 	]
> > );
> > 
> > $this->add_control(
> > 	'list',
> > 	[
> > 		'label' => __( 'Repeater List', 'plugin-domain' ),
> > 		'type' => \Elementor\Controls_Manager::REPEATER,
> > 		'fields' => $repeater->get_controls(),
> > 		'default' => [
> > 			[
> > 				'list_title' => __( 'Title #1', 'plugin-domain' ),
> > 				'list_content' => __( 'Item content. Click the edit button to change this text.', 'plugin-domain' ),
> > 			],
> > 			[
> > 				'list_title' => __( 'Title #2', 'plugin-domain' ),
> > 				'list_content' => __( 'Item content. Click the edit button to change this text.', 'plugin-domain' ),
> > 			],
> > 		],
> > 		'title_field' => '{{{ list_title }}}',
> > 	]
> > );
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Official Documentation: https://developers.elementor.com/elementor-controls/repeater-control/

Thanks mate. That solution worked perfectly

**Comment by tcj-rocks** [comment]
Almost three years after this has been reported, the issue is still present.
